src/modules/module_29600.c

```diff
@@ -171,7 +184,52 @@ int module_hash_decode (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSE
 
 int module_hash_encode (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED const void *digest_buf, MAYBE_UNUSED const salt_t *salt, MAYBE_UNUSED const void *esalt_buf, MAYBE_UNUSED const void *hook_salt_buf, MAYBE_UNUSED const hashinfo_t *hash_info, char *line_buf, MAYBE_UNUSED const int line_size)
 {
-  const int line_len = snprintf (line_buf, line_size, "%s", hash_info->orighash);
+  terra_t *terra = (terra_t *) esalt_buf;
+
+  u32 *digest = (u32 *) digest_buf;
+
+  // salt:
+
+  char salt_hex[32 + 1] = { 0 };
+
+  hex_encode ((u8 *) terra->salt_buf, 16, (u8 *) salt_hex); // or use: generic_salt_encode ()
+
+  salt_hex[32] = 0;
+
+  // iv:
+
+  char iv_hex[32 + 1] = { 0 };
+
+  hex_encode ((u8 *) terra->iv, 16, (u8 *) iv_hex);
```

Comment:
I am thinking about adding some `TERRA_SALT_LEN` and `TERRA_SALT_HEX_LEN` like here

https://github.com/hashcat/hashcat/blob/master/src/modules/module_29311.c#L49-L53

https://github.com/hashcat/hashcat/blob/master/src/modules/module_29311.c#L233-L237

Reply:
OK